src/libasr/asr_utils.h

```diff
@@ -105,6 +105,33 @@ static inline ASR::ttype_t* symbol_type(const ASR::symbol_t *f)
     return nullptr;
 }
 
+static inline ASR::ttype_t* get_contained_type(ASR::ttype_t* asr_type) {
+    switch( asr_type->type ) {
+        case ASR::ttypeType::List: {
+            return ASR::down_cast<ASR::List_t>(asr_type)->m_type;
+        }
+        case ASR::ttypeType::Set: {
+            return ASR::down_cast<ASR::Set_t>(asr_type)->m_type;
+        }
+        case ASR::ttypeType::Enum: {
+            ASR::Enum_t* enum_asr = ASR::down_cast<ASR::Enum_t>(asr_type);
+            ASR::EnumType_t* enum_type = ASR::down_cast<ASR::EnumType_t>(enum_asr->m_enum_type);
+            return enum_type->m_type;
+        }
+        case ASR::ttypeType::Pointer: {
+            ASR::Pointer_t* pointer_asr = ASR::down_cast<ASR::Pointer_t>(asr_type);
+            return pointer_asr->m_type;
+        }
+        case ASR::ttypeType::Const: {
+            ASR::Const_t* const_asr = ASR::down_cast<ASR::Const_t>(asr_type);
+            return const_asr->m_type;
+        }
+        default: {
+            return asr_type;
+        }
```

Comment:
What if it is a `Const[Pointer[i32]]`?

Reply:
For now, let's add (in subsequent PRs) tests to verify() that only single level of nesting is allowed for Const and Pointer.

Reply:
> What if it is a `Const[Pointer[i32]]`?

Well, in that case we will receive, `Pointer[i32]` which is what we need? Is there any case where we need `i32` out of `Const[Pointer[i32]]`. In fact we can specify the depth upto which we want to unwrap a given type. But yeah that should be done when we find the need for it via a concrete case. I will open an issue for this.